controllers/events_controller.js

```diff
@@ -128,26 +130,28 @@ module.exports.registerParticipant = async (req, res) => {
 };
 
 module.exports.updateParticipant = async (req, res) => {
-	let { name, email, branch, year, phone, password } = req.body;
-	let updateObj = {
-		name,
-		email,
-		branch,
-		year,
-		phone,
-		password
-	};
+	let { name, email } = req.body;
+
+	let participant = await Participant.findById(req.params.id);
+
+	if (!participant)
+		return sendError(res, "Participant not found!!", BAD_REQUEST);
+
+	if (name && name !== participant.name) {
+		setToken(req.query.id, "revalidate");
+	}
+
+	if (email && email !== participant.email) {
```

Comment:
do we need email verification on email change?

Reply:
No, token will get renewed only when the user logins again. So if the email provided is correct then they'll login easily

Reply:
okay